editor/js/Editor.js

```diff
@@ -234,6 +236,32 @@ Editor.prototype = {
 	addMaterial: function ( material ) {
 
 		this.materials[ material.uuid ] = material;
+		this.signals.materialAdded.dispatch();
+
+	},
+
+	removeMaterial: function ( material ) {
+
+		delete this.materials[ material.uuid ];
+		this.signals.materialRemoved.dispatch();
+
+	},
+
+	getMaterialById ( id ) {
+
+		var material;
+		var materials = Object.values( this.materials );
+
+		for( var i = 0; i < materials.length; i ++ ) {
+			
+			if ( materials[i].id === id ) {
```

Comment:
There are some minor style issues in your code. I suggest you setup your editor in a way so it automatically respect the project's style guide (defined in [mdcs](https://github.com/mrdoob/three.js/blob/2a8aa96d69c6a2e7daf4689db95f136f9164adb8/package.json#L79)).

You can then automatically fix such issues and produce outputs like:
```js
if ( materials[ i ].id === id ) {

    material = materials[ i ];
    break;

}

Reply:
Will do it,

Will need some getting used to, to these linters :)